ops/terrad/terrad.go

```diff
@@ -353,28 +371,162 @@ func (t Terrad) InitOCR(keys []opsChainlink.NodeKeys) error {
 		transmitterArray = append(transmitterArray, string(transmitters[i]))
 	}
 
-	tx := SetConfig{
-		SetConfig: SetConfigDetails{
-			Signers:               signerArray,
-			Transmitters:          transmitterArray,
-			F:                     f,
-			OnchainConfig:         onchainConfig,
+	status = utils.LogStatus("InitOCR: begin proposal")
+	resp, err := t.executeOCR2(BeginProposal)
+	if err == nil {
+		if resp.Code != 0 {
+			err = fmt.Errorf("tx response contains error: %s %d", resp.Codespace, resp.Code)
+		} else if len(resp.Logs) == 0 {
+			err = errors.New("begin proposal produced no logs")
+		}
+	}
+	if status.Check(err) != nil {
+		return err
+	}
+
+	var id string
+	for _, e := range resp.Logs[0].Events {
+		if e.Type == "wasm" {
+			for _, a := range e.Attributes {
+				if a.Key == "proposal_id" {
+					if a.Value == "" {
+						return errors.New("empty proposal id")
+					}
+					id = a.Value
+				}
+			}
+		}
+	}
+	if id == "" {
+		return errors.New("failed to find event with attribute: wasm.proposal-id")
+	}
+
+	defer func() {
```

Comment:
Proposals are cleared after being accepted, so not really necessary unless the idea is to clear broken proposals on error. In that case, an additional comment would be appreciated, about why the code is here.